dovecot.service.in

```diff
@@ -30,8 +30,10 @@ ExecReload=@bindir@/doveadm reload
 ExecStop=@bindir@/doveadm stop
 PrivateTmp=true
 NonBlocking=yes
-# Enable this if your systemd is new enough to support it:
-#ProtectSystem=full
+ProtectSystem=full
+NoNewPrivileges=true
+PrivateDevices=true
+CapabilityBoundingSet=CAP_CHOWN CAP_DAC_OVERRIDE CAP_IPC_LOCK CAP_KILL CAP_NET_BIND_SERVICE CAP_SETGID CAP_SETUID CAP_SYS_CHROOT
```

Comment:
@cmouse I don't think `NoNewPrivileges` prevents UID changes - it prevents increasing privileges once they've been lost. If, for example, dovecot starts as root, then drops privileges and changes UID to `dovecot`, that would be fine. If, on the other hand, dovecot starts as root, then drops privileges and changes UID to `dovecot`, then tries to regain dropped privileges, then `NoNewPrivileges` would block that.

I'm curious - is there a problem that you're seeing? I'd be really interested in knowing what the problem may be.